File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    box = []
    stack = []
    for i in range(len(board)):
        b = []
        for j in range(len(board)):
            if board[j][i] != 0:
                b.append(board[j][i])
        box.append(b)
   
    for i in moves:
        if not box[i-1]:
            continue
        if not stack or stack[-1] != box[i-1][0]:
            stack.append(box[i-1][0])
        else:
            stack.pop()
            answer += 1
        box[i-1].pop(0)        
    return answer * 2

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        List<Integer> stack = new ArrayList<>();
        
        
        for(int i : moves) {
            for(int j = 0; j < board.length; j++){
                if(board[j][i-1] != 0) {
                    stack.add(board[j][i-1]);
                    board[j][i-1] = 0;
                    
                    if (stack.size() > 1) {
                        if (stack.get(stack.size()-1) == stack.get(stack.size()-2)) {
                            stack.remove(stack.size()-1);
                            stack.remove(stack.size()-1);
                            answer += 2;
                        }
                    }
                    break;                    
                }
            }
        }
        
        return answer;
    }
}